pageserver/src/walrecord.rs

```diff
@@ -270,13 +270,59 @@ pub struct XlHeapDelete {
 
 impl XlHeapDelete {
     pub fn decode(buf: &mut Bytes) -> XlHeapDelete {
-        XlHeapDelete {
-            xmax: buf.get_u32_le(),
-            offnum: buf.get_u16_le(),
-            _padding: buf.get_u16_le(),
-            t_cid: buf.get_u32_le(),
-            infobits_set: buf.get_u8(),
-            flags: buf.get_u8(),
+        if buf.remaining() == 14 {
+            XlHeapDelete {
```

Comment:
```suggestion
        if buf.remaining() == 14 {
            // since pg 16
            XlHeapDelete {
```

I think it would be healthy to add such comments, but we don't have such comments yet. What do you think, is this without one (in all added places) fine as long as we can blame for it.

I just noticed this is not for pg16 but for ... neon preprocessed wal?

Reply:
14 (right now renamed to pg_constants::SIZE_OF_HEAP_DELETE) is unsigned size of HeapDelete WAL record with t_cid added by Neon. Size of Vanilla structure is 6 bytes smaller. Certainly format of this WAL record can be changed in future Postgres releases. We should carefully maintain this code.

In all other places I rely on flags. But in this two places (heap_remove and heap_delete) `flags` field is extracted after `t_cid`. So I have to check size of WAL record.

Reply:
Allright..